addons/l10n_cl/models/account_move.py

```diff
@@ -30,25 +30,26 @@ def _get_l10n_latam_documents_domain(self):
                 document_type_ids = self.journal_id.l10n_cl_sequence_ids.mapped('l10n_latam_document_type_id').ids
             else:  # self.journal_id.type == 'purchase':
                 partner_domain = [
-                        ('country_id.code', '=', 'CL'),
-                        ('internal_type', 'in', ['invoice', 'debit_note', 'credit_note']), ('active', '=', True)]
+                    ('country_id.code', '=', 'CL'),
+                    ('internal_type', 'in', ['invoice', 'debit_note', 'credit_note', 'invoice_in']),
+                    ('active', '=', True)]
                 if len(self.partner_id) == 0:
-                    # this will allow to use the whole documents domain if partner not selected.
                     pass
-                elif self.partner_id.l10n_cl_sii_taxpayer_type == '1':
+                elif self.partner_id.l10n_cl_sii_taxpayer_type == '1' and self.partner_id_vat != '60805000-0':
+                    partner_domain += [('code', 'not in', ['39', '70', '71', '914', '911'])]
+                elif self.partner_id.l10n_cl_sii_taxpayer_type == '1' and self.partner_id_vat == '60805000-0':
                     partner_domain += [('code', 'not in', ['39', '70', '71'])]
                 elif self.partner_id.l10n_cl_sii_taxpayer_type == '2':
-                    partner_domain += [('code', 'in', ['70', '71', '56', '61'])]
-                elif self.partner_id_vat == '60805000-0':
-                    partner_domain += [('code', '=', '914')]
+                    partner_domain += [('code', 'in', ['70', '71', '56', '61']), ('code', 'not in', ['914', '911'])]
                 elif self.partner_id.l10n_cl_sii_taxpayer_type == '3':
-                    partner_domain += [('code', 'in', ['35', '38', '39', '41', '56', '61'])]
+                    partner_domain += [('code', 'in', ['35', '38', '39', '41', '56', '61']),
+                                       ('code', 'not in', ['914', '911'])]
                 elif not self.partner_id.l10n_cl_sii_taxpayer_type or self.partner_id.country_id != self.env.ref(
                         'base.cl') or self.partner_id.l10n_cl_sii_taxpayer_type == '4':
                     partner_domain += [('code', 'in', [])]
```

Comment:
We find this very convenient. Further validation is just in case for some reason the filter is overridden (for example changing some other value, as the journal). Nevertheless, later we put an onchange that try to address this case also.

Reply:
Fixed code in / code not in at
ccad6d8f3bb8732756a53adc260245de9eee06b4